apps/ledger-live-desktop/tests/page/speculos.page.ts

```diff
@@ -16,25 +17,35 @@ import { Currency } from "tests/enum/Currency";
 export class SpeculosPage extends AppPage {
   @step("Verify receive address correctness")
   async expectValidReceiveAddress(account: Account) {
-    await waitFor(account.currency.receivePattern[0]);
-    const actualAddress = await pressRightUntil(account.currency.receivePattern[0]);
+    await waitFor(account.currency.speculosApp.receivePattern[0]);
+    const actualAddress = await pressRightUntil(account.currency.speculosApp.receivePattern[0]);
     expect(assertAddressesEquality(account.address, actualAddress)).toBe(true);
-    await pressRightUntil(account.currency.receivePattern[1]);
+    await pressRightUntil(account.currency.speculosApp.receivePattern[1]);
     await pressBoth();
   }
 
   @step("Verify transaction info on device")
   async expectValidTxInfo(tx: Transaction) {
-    const amountScreen = await pressRightUntil(tx.accountToDebit.currency.sendPattern[0]);
+    const amountScreen = await pressRightUntil(
+      tx.accountToDebit.currency.speculosApp.sendPattern[0],
+    );
     expect(verifyAmount(tx.amount, amountScreen)).toBe(true);
-    const addressScreen = await pressRightUntil(tx.accountToDebit.currency.sendPattern[1]);
+    const addressScreen = await pressRightUntil(
+      tx.accountToDebit.currency.speculosApp.sendPattern[1],
+    );
     expect(assertAddressesEquality(tx.accountToCredit.address, addressScreen)).toBe(true);
-    await pressRightUntil(tx.accountToDebit.currency.sendPattern[2]);
+    await pressRightUntil(tx.accountToDebit.currency.speculosApp.sendPattern[2]);
     await pressBoth();
     if (tx.accountToDebit.currency.name === Currency.DOGE.name) {
       await waitFor("Confirm");
       await pressRightUntil(DeviceLabels.ACCEPT);
       await pressBoth();
     }
   }
+
+  @step("Verify the screen displays the expected text")
+  async confirmOperationOnDevice(text: string) {
```

Comment:
the name of the method and step description are misleading, what actually happens here? confirmation or text verification?